pkg/agent/config/node_config.go

```diff
@@ -265,23 +268,33 @@ func (nc *NetworkConfig) NeedsDirectRoutingToPeer(peerIP net.IP, localIP *net.IP
 }
 
 func (nc *NetworkConfig) CalculateMTUDeduction(isIPv6 bool) int {
-	var mtuDeduction int
+	// When WireGuard is enabled, NetworkConfig.TunnelType will be ignored, so we deduct MTU based on WireGuardOverhead.
+	if nc.TrafficEncryptionMode == TrafficEncryptionModeWireGuard {
```

Comment:
I suppose we need to reduce ipv6ExtraOverhead as well when the WireGuard is enabled?  @tnqn 

You probably can move lines 290-292 before  271.

Reply:
Moved the IPv6 check to above of WG check